color/command.c

```diff
@@ -639,7 +638,7 @@ static enum CommandResult parse_color(struct Buffer *buf, struct Buffer *s,
     {
       struct Buffer tmp = mutt_buffer_make(0);
       mutt_extract_token(&tmp, s, MUTT_TOKEN_NO_FLAGS);
-      if (mutt_str_atoui(tmp.data, &match) < 0)
+      if (!mutt_str_atoui(tmp.data, &match))
```

Comment:
`mutt_str_atoui_full()`

Reply:
I agree, but this is changing the current behaviour, where `color status red default '\[([^]]+)\]' 1foobar` works.

Reply:
I vote for fixing the behaviour.
I can't imagine anyone relying on this mis-parsing.

(`color status fg bg pattern` is a NeoMutt extension, so we haven't inherited the code)